src/Razor/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/CanProduceLinePragmasForComponentWithRenderFragment/TestComponent.codegen.cs

```diff
@@ -0,0 +1,88 @@
+// <auto-generated/>
+#pragma warning disable 1591
+namespace Test
+{
+    #line hidden
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Components;
+    public partial class TestComponent : Microsoft.AspNetCore.Components.ComponentBase
+    {
+        #pragma warning disable 219
+        private void __RazorDirectiveTokenHelpers__() {
+        }
+        #pragma warning restore 219
+        #pragma warning disable 0414
+        private static System.Object __o = null;
+        #pragma warning restore 0414
+        #pragma warning disable 1998
+        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder)
+        {
+#nullable restore
+#line 2 "x:\dir\subdir\Test\TestComponent.cshtml"
+                                        __o = ActionText;
+
+#line default
+#line hidden
+#nullable disable
+#nullable restore
+#line 3 "x:\dir\subdir\Test\TestComponent.cshtml"
+   if (!Collapsed)
+  {
+    
+
+#line default
+#line hidden
+#nullable disable
+#nullable restore
+#line 6 "x:\dir\subdir\Test\TestComponent.cshtml"
+ __o = ChildContent;
+
+#line default
+#line hidden
+#nullable disable
+#nullable restore
+#line 7 "x:\dir\subdir\Test\TestComponent.cshtml"
+          
+  }
+
+#line default
+#line hidden
+#nullable disable
+        }
+        #pragma warning restore 1998
+#nullable restore
+#line 11 "x:\dir\subdir\Test\TestComponent.cshtml"
+ 
+  [Parameter]
+  public RenderFragment ChildContent { get; set; } = (context) => 
+
+#line default
+#line hidden
+#nullable disable
+#nullable restore
+#line 13 "x:\dir\subdir\Test\TestComponent.cshtml"
```

Comment:
Interesting that we're not generating enhanced line pragmas at design time too. That being said it's probably the best way to start

Reply:
Ah, this was intentional on my part. I thought we agreed to only limit enhanced line pragmas to scenarios where it was necessary and supported (>= C# 10 and files that are used for diagnostics/debugging).

Reply:
Ya it's fine for now, I hadn't thought it all through is all 😄